dask_geomodeling/raster/spatial.py

```diff
@@ -429,3 +435,172 @@ def get_sources_and_requests(self, **request):
+        except RuntimeError:
+            raise ValueError(
+                "'{}' is not a valid projection string".format(place_projection)
+            )
+        anchor = list(anchor)
+        if len(anchor) != 2:
+            raise ValueError("Expected 2 numbers in the 'anchor' parameter")
+        for x in anchor:
+            if not isinstance(x, (int, float)):
+                raise TypeError("'{}' object is not allowed".format(type(x)))
+        coordinates = np.asarray(coordinates, dtype=float)
+        if coordinates.ndim != 2 or coordinates.shape[1] != 2:
+            raise ValueError(
+                "Expected a list of lists of 2 numbers in the 'coordinates' "
+                "parameter"
+            )
+        super().__init__(store, place_projection, anchor, coordinates.tolist())
+
+    @property
+    def place_projection(self):
+        return self.args[1]
+
+    @property
+    def anchor(self):
+        return self.args[2]
+
+    @property
+    def coordinates(self):
+        return self.args[3]
+
+    @property
+    def projection(self):
+        """The native projection of this block.
+
+        Only returns something if the place projection equals the store
+        projection"""
+        store_projection = self.store.projection
+        if store_projection is None:
+            return
+        if get_sr(self.place_projection).IsSame(get_sr(store_projection)):
+            return store_projection
+
+    @property
+    def geo_transform(self):
+        """The native geo_transform of this block
+
+        Returns None if the store projection and place projections differ."""
+        if self.projection is not None:
+            return self.store.geo_transform
+
+    @property
+    def extent(self):
+        geometry = self.geometry
+        if not geometry.GetSpatialReference().IsSame(EPSG4326):
+            geometry = geometry.Clone()
+            geometry.TransformTo(EPSG4326)
+        x1, x2, y1, y2 = geometry.GetEnvelope()
+        return x1, y1, x2, y2
+
+    @property
+    def geometry(self):
+        """Combined geometry in this block's native projection. """
+        store_geometry = self.store.geometry
+        sr = get_sr(self.place_projection)
+        if not store_geometry.GetSpatialReference().IsSame(sr):
+            store_geometry = store_geometry.Clone()
+            store_geometry.TransformTo(sr)
+        x1, x2, y1, y2 = store_geometry.GetEnvelope()
+
+        extents = []
+        for _x, _y in self.coordinates:
+            extents.append(
+                [
+                    x1 - self.anchor[0] + _x,
+                    y1 - self.anchor[1] + _y,
+                    x2 - self.anchor[0] + _x,
+                    y2 - self.anchor[1] + _y,
+                ]
+            )
+
+        # join the extents
+        x1 = min([e[0] for e in extents])
+        y1 = min([e[1] for e in extents])
+        x2 = max([e[2] for e in extents])
+        y2 = max([e[3] for e in extents])
```

Comment:
I couldn't resist. What do you think?

```suggestion
         _x1, _x2, _y1, _y2 = store_geometry.GetEnvelope()
         p, q = self.anchor
         P, Q = zip(*self.coordinates)
         x1, x2 = _x1 + min(P) - p, _x2 + max(P) - p
         y1, y2 = _y1 + min(Q) - q, _y2 + max(Q) - q
```